README.md

```diff
@@ -43,4 +42,119 @@ The JVM will then listen for a debugger to be attached on port 8189.  The JVM wi
+        user: echo_migrate
+    ```
+
+See [Sample deployment topology](#sample-deployment-topology) for additional information
+
+### Configuration options
+`echo` has several configuration options (can be specified in `echo.yml`), key ones are listed below:  
+* `scheduler.enabled` (default: `false`)  
+    when set to `true` this instance will schedule and trigger CRON events
+* `scheduler.pipelineConfigsPoller.enabled` (default: `false`)  
+    when `true`, will poll `front50` for pipeline configurations and sync triggers (you should definitely set this to `true` if you enable `scheduler`)
+* `scheduler.compensationJob.enabled` (default: false)  
+    when `true` this instance will poll for missed CRON triggers and attempt to re-trigger them (see [Missed CRON scheduler](#Missed-CRON-scheduler))
+* `orca.pipelineInitiatorRetryCount` (default: `5`)  
+    Number of retries on `orca` failures (leave at default)
+* `orca.pipelineInitiatorRetryDelayMillis` (default: 5000ms)
+    Number of milliseconds between retries to `orca` (leave at default)
+
+## Missed CRON scheduler
+The missed CRON scheduler is a feature in `echo` that ensures that CRON triggers are firing reliably.  
+In an event that a CRON trigger fails to fire or it fires but orca fails to start the pipeline the missed CRON scheduler will detect it and attempt to re-trigger the pipeline.  
+The main scenario when missed cron scheduler is 
+
+## Sample deployment topology
+Here are examples of what configurations you can deploy `echo`.
+
+|                   | Using in-memory           | Using SQL |
+|-------------------|---------------------------|-----------|
+|**Server Group 1** |3x `echo`                  | 3x `echo` with `echo-scheduler`
+|**Server Group 2** |1x `echo-scheduler`*       | 1x `echo-missed-scheduler`*
+|**Server Group 3** |1x `echo-missed-scheduler` | n/a
+
+\* _optional `echo-missed-scheduler` see [Missed CRON scheduler](#Missed-CRON-scheduler)_
+
+If you opt for using an in-memory execution mode, take care when deploying upgrades to `echo`.
+Since only instance should be running at a time, a rolling-push strategy will need to be used. Furthermore, if using `echo-missed-scheduler`, make sure to upgrade `echo-scheduler` followed by `echo-missed-scheduler` to ensure pipelines (which had a trigger during the deploy period) are re-triggered correctly after deploy.
+
+The following are configuration options for each server group (note that other configurations options will be requires, which halyard will configure):  
+`echo`  
+```yaml
+scheduler:
+  enabled: false
+  pipelineConfigsPoller:
+    enabled: false
+  compensationJob:
+    enabled: false
+```
+
+`echo-scheduler`  
+```yaml
+scheduler:
+  enabled: true
+  pipelineConfigsPoller:
+    enabled: true
+  compensationJob:
+    enabled: false
+```
+
+`echo-missed-scheduler`  
+```yaml
+scheduler:
+  enabled: true
+  pipelineConfigsPoller:
+    enabled: false
```

Comment:
Above it says this should definitely be `true` if the scheduler is enabled

Reply:
tweaking :)